### PR TITLE
7800.main: Migrated data from d7

### PR DESCRIPTION
**Pull request by faust174**

### CODE REVIEW

File: web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Removes links from body if they match a specific URL.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "custom_link_plugin"
+ * )
+ */
+class CustomLinkPlugin extends ProcessPluginBase {
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
```
Comment (KuzyaWkk) [inline web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php]: Missed comment

File: web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php
Comment (KuzyaWkk) [inline web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php:1]: I'm not sure this is the right way to do things

**Review by KuzyaWkk — CHANGES_REQUESTED** [review]
Add comment to function transform

**Review by KuzyaWkk — APPROVED** [review]

File: web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    $callback = function ($matches) {
+      $url = $matches[2];
+      $modified_url = str_replace('ehistory.osu.edu', '', $url);
+      return $matches[1] . $modified_url . $matches[3];
+    };
+
+    $pattern = '/(<a[^>]+href=["\'])([^"\']+ehistory\.osu\.edu\/[^"\']*)(["\'][^>]*>)/i';
+
+    return preg_replace_callback($pattern, $callback, $value);
```
Comment (lavrenchuk-dev) [inline web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php]: There are libraries / helpers to work with the HTML
regex magic is not needed (not explicitly at least)

**Review by lavrenchuk-dev — CHANGES_REQUESTED** [review]

File: web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php [hunk 3]
```diff
@@ -14,20 +14,29 @@
...
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property): array
+  {
+    if (!empty($value['value'])) {
+      $dom = new \DOMDocument();
+      $dom->loadHTML($value['value']);
+      $anchors = $dom->getElementsByTagName('a');
+      foreach ($anchors as $anchor) {
+        $href = $anchor->getAttribute('href');
+        $href = str_replace('http://ehistory.osu.edu', '', $href);
+        $is_valid = \Drupal::pathValidator()->isValid($href);
```
Comment (faust174) [inline web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php]: Didn't use DI here cause faced an error, was not able to fix it fast and didn't want to spend too much time fixing it. So left it as it is.

Error message: TypeError: Drupal\migration\Plugin\migrate\process\CustomLinkPlugin::__construct(): Argument #4 ($pathValidator) must be of type Drupal\Core\Path\PathValidatorInterface, Drupal\node\Plugin\migrate\D7NodeTranslation given

**Review by lavrenchuk-dev — APPROVED** [review]

**Comment by lavrenchuk-dev** [comment]
@faust174 I have no rights to merge PRs here
Feel free to merge yourself

**Pull request merged**